common/src/main/java/com/wynntils/features/chat/ChatMentionFeature.java

```diff
@@ -77,32 +99,34 @@ public void onChat(ChatMessageReceivedEvent e) {
             }
 
             StyledTextPart partToReplace = part;
-            Matcher matcher = mentionPattern.matcher(partToReplace.getString(null, PartStyle.StyleType.NONE));
-
-            while (matcher.find()) {
-                String match = partToReplace.getString(null, PartStyle.StyleType.NONE);
-
-                String firstPart = match.substring(0, matcher.start());
-                String mentionPart = match.substring(matcher.start(), matcher.end());
-                String lastPart = match.substring(matcher.end());
-
-                PartStyle partStyle = partToReplace.getPartStyle();
-
-                StyledTextPart first = new StyledTextPart(firstPart, partStyle.getStyle(), null, Style.EMPTY);
-                StyledTextPart mention = new StyledTextPart(
-                        mentionPart,
-                        partStyle.getStyle().withColor(mentionColor.get().getChatFormatting()),
-                        null,
-                        first.getPartStyle().getStyle());
-                StyledTextPart last = new StyledTextPart(lastPart, partStyle.getStyle(), null, Style.EMPTY);
-
-                changes.remove(partToReplace);
-                changes.add(first);
-                changes.add(mention);
-                changes.add(last);
-
-                partToReplace = last;
-                matcher = mentionPattern.matcher(lastPart);
+            for (Pattern pattern : mentionPatterns) {
+                Matcher matcher = pattern.matcher(partToReplace.getString(null, PartStyle.StyleType.NONE));
+
+                while (matcher.find()) {
+                    String match = partToReplace.getString(null, PartStyle.StyleType.NONE);
+
+                    String firstPart = match.substring(0, matcher.start());
+                    String mentionPart = match.substring(matcher.start(), matcher.end());
+                    String lastPart = match.substring(matcher.end());
+
+                    PartStyle partStyle = partToReplace.getPartStyle();
+
+                    StyledTextPart first = new StyledTextPart(firstPart, partStyle.getStyle(), null, Style.EMPTY);
```

Comment:
@kristofbolyai Btw, this code here is very close to the functionality needed in the PR about trade market item names. I think we have an opportunity to make some more general functionality in StyledText to manipulate styled texts based on substrings.

Perhaps if we had a version of split that also returned the part that matched the split. So here we could have done like this pseudo-code:
```
                    StyledText[] splitStyledText = partToReplace.splitAndKeep(matcher.group());

                    changes.remove(partToReplace);
                  
                    changes.add(splitStyledText[0]);
                    changes.add(new StyledText(splitStyledText[1].getString(), new Style(mentionColor.get());
                    changes.add(splitStyledText[2]);

```

Reply:
That would be a bit different than split, but I see what you are getting at. I plan to look at a styled text issue later today, maybe I check this too. But I also have a lot open so..